src/impl/peerconnection.cpp

```diff
@@ -45,9 +45,17 @@ static LogCounter
                                 "Number of unknown RTCP packet types over past second");
 
 PeerConnection::PeerConnection(Configuration config_)
-    : config(std::move(config_)), mCertificate(make_certificate(config.certificateType)) {
+    : config(std::move(config_)) {
 	PLOG_VERBOSE << "Creating PeerConnection";
 
+	if( config.certPem.has_value() && config.keyPem.has_value() ) {
+		std::promise<certificate_ptr> cert;
+		cert.set_value(std::make_shared<Certificate>(Certificate::FromString(config.certPem.value(), config.keyPem.value())));
+		mCertificate = cert.get_future();
+	} else {
```

Comment:
You should create the certificate if both certificate and key are not specified, else throw an `invalid_argument` exception (like in [WebSocketServer](https://github.com/paullouisageneau/libdatachannel/blob/3a11fec497921fcb74c29c27d6e01b7071539c42/src/impl/websocketserver.cpp#L36])).

Reply:
get it.